pkg/testing/be_sematic_equal_matcher.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"fmt"
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/types"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"reflect"
+)
+
+// Gomega does not support semantic equal so adding this here in testing util
+
+func BeSematicEqual(expected interface{}) types.GomegaMatcher {
```

Comment:
How about Should(SemanticallyEqual())?

Reply:
this is gomega's naming convention, going to chat with them to see if it is possible to upstream this.